Description: Don't change uart speed before firmware load
 TODO: Put a short summary on the line above and replace this paragraph
 with a longer explanation of this change. Complete the meta-information
 with other relevant fields (see below for details). To make it easier, the
 information below has been extracted from the changelog. Adjust it or drop
 it.
 .
 bluez (5.43-2+deb9u1) stretch-security; urgency=high
 .
   * Non-maintainer upload by the Security Team.
   * CVE-2017-1000250: information disclosure vulnerability in
     service_search_attr_req (Closes: #875633)
Author: Salvatore Bonaccorso <carnil@debian.org>
Bug-Debian: https://bugs.debian.org/875633

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: 2018-11-26

--- bluez-5.43.orig/tools/hciattach_bcm43xx.c
+++ bluez-5.43/tools/hciattach_bcm43xx.c
@@ -368,8 +368,8 @@ int bcm43xx_init(int fd, int def_speed,
 	if (bcm43xx_locate_patch(FIRMWARE_DIR, chip_name, fw_path)) {
 		fprintf(stderr, "Patch not found, continue anyway\n");
 	} else {
-		if (bcm43xx_set_speed(fd, ti, speed))
-			return -1;
+		//if (bcm43xx_set_speed(fd, ti, speed))
+		//	return -1;
 
 		if (bcm43xx_load_firmware(fd, fw_path))
 			return -1;
@@ -379,6 +379,7 @@ int bcm43xx_init(int fd, int def_speed,
 			perror("Can't set host baud rate");
 			return -1;
 		}
+		sleep(1);
 
 		if (bcm43xx_reset(fd))
 			return -1;
